Fix for #7149 + proposed readme for docker on Windows + fixing typos

joseph-sch opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Type

<!--- Select with 'x' and link to a related issue. What types of changes does your code introduce? -->

-   [x] Bug fix (non-breaking change which fixes an issue): Fixes #7149
-   [ ] New feature (non-breaking change which adds functionality). Resolves #
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) Resolves #7149

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

1. There is a typo in `o3d.geometry.TriangleMesh.create_mobius` and in `o3d.t.geometry.TriangleMesh.create_mobius`: `raidus` instead of `radius` as explained in #7149
2. Some typos throughout the docstrings, comments and a string
3. Proposing a markdown readme for docker on Windows - or should it be part of the `docker.in.rst` file in order to appear within https://www.open3d.org/docs/latest/docker.html#building-open3d-in-docker?

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that
apply.  If you're unsure about any of these, don't hesitate to ask. We're here
to help! -->

-   [x] I have run `python util/check_style.py --apply` to apply Open3D **code style**
    to my code.
-   [ ] This PR changes Open3D behavior or adds new functionality.
    -   [ ] Both C++ (Doxygen) and Python (Sphinx / Google style) **documentation** is
        updated accordingly.
    -   [ ] I have added or updated C++ and / or Python **unit tests** OR included **test
        results** (e.g. screenshots or numbers) here.
-   [x] I will follow up and update the code if CI fails.
    <!-- In case I am unavailable later -->
-   [x] For fork PRs, I have selected **Allow edits from maintainers**.

## Description

<!--- Describe your changes, with test results and screenshots as appropriate. Move unrelated changes, if any, to a separate PR. -->

timohl reviewed (approved):
All good small fixes.

I would mark this PR as breaking change though, since a parameter name changed (with good reason).
It is just a typo fix, but existing calls using named parameters would break.

I cannot say much to the content of the docker windows readme, but there is also [docker/README.md](https://github.com/isl-org/Open3D/blob/main/docker/README.md) where this could be included as a section (or in [docs/docker.in.rst](https://github.com/isl-org/Open3D/blob/main/docs/docker.in.rst) as you mentioned.
Not sure if it is a good idea to have 3 different spots documenting docker builds.

joseph-sch commented:
> All good small fixes.
> 
> I would mark this PR as breaking change though, since a parameter name changed (with good reason). It is just a typo fix, but existing calls using named parameters would break.
> 
> I cannot say much to the content of the docker windows readme, but there is also [docker/README.md](https://github.com/isl-org/Open3D/blob/main/docker/README.md) where this could be included as a section (or in [docs/docker.in.rst](https://github.com/isl-org/Open3D/blob/main/docs/docker.in.rst) as you mentioned. Not sure if it is a good idea to have 3 different spots documenting docker builds.

Thank you for the feedback 
1. indeed technically it's a breaking change, I've updated the description
2. I can split the instructions that I proposed between a short mention in [docs/docker.in.rst](https://github.com/isl-org/Open3D/blob/main/docs/docker.in.rst) and detailed instructions in [docker/README.md](https://github.com/isl-org/Open3D/blob/main/docker/README.md) - would that be OK?

timohl commented:
> 2\. I can split the instructions that I proposed between a short mention in [docs/docker.in.rst](https://github.com/isl-org/Open3D/blob/main/docs/docker.in.rst) and detailed instructions in [docker/README.md](https://github.com/isl-org/Open3D/blob/main/docker/README.md) - would that be OK?

Something like that sounds good I think, but somebody else should review that, since I am not using open3d+docker.

ssheorey reviewed (changes requested):
Hi @joseph-sch , this is very helpful. Thanks! I can't believe we didn't catch that typo before! Can I recommend merging the WSL docker documentation to the current docker docs in `docker/README.md`.  It's easier for users if it's in a single file.

joseph-sch commented:
> Hi @joseph-sch , this is very helpful. Thanks! I can't believe we didn't catch that typo before! Can I recommend merging the WSL docker documentation to the current docker docs in `docker/README.md`. It's easier for users if it's in a single file.

Yes indeed - I’m aware that I have to do that but I didn’t get around to doing it yet. I’ll try to do it in the coming 48 hours.

joseph-sch commented:
I have removed the extra `.md` file (`BUILD_DOCKER_WSL.md`) and incorporated the contents as clarifications in `docker.in.rst` and in the `docker/README.md` file. I hope it's fine but of course a review would be good. 

Besides, I see there is a conflict to be resolved in `cpp/pybind/t/geometry/trianglemesh.cpp`.
